### PR TITLE
Enable OMP runtime on windows, add Threadpool runtime on all OSs

### PR DESCRIPTION
**Pull request by igorsafo**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Promote the changes discussed in the following proposal about conda-forge distribution: [#1054](https://github.com/oneapi-src/oneDNN/pull/1054)
- Add OMP runtime on Windows
- Add Threadpool runtime on all OSs

### CODE REVIEW

**Comment by igorsafo** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 dnnl_cpu_runtime:
   - tbb  # [not (ppc64le or aarch64)]
   - omp  # [unix]
+  - threadpool # [not (ppc64le or aarch64)]
```
Comment (xhochy) [inline recipe/conda_build_config.yaml:4]: This should probably also be skipped for macOS ARM?
```suggestion
  - threadpool # [not (ppc64le or aarch64 or arm64)]
```

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
 dnnl_cpu_runtime:
   - tbb  # [not (ppc64le or aarch64)]
   - omp  # [unix]
```
Comment (xhochy) [inline recipe/conda_build_config.yaml:3]: If you want to add OpenMP for Windows, you need to remove the `unix` here and rerender.
```suggestion
  - omp
```

File: recipe/bld-onednn.bat [hunk 3]
```diff
@@ -2,11 +2,17 @@
...
 md "%SRC_DIR%"\build
 pushd "%SRC_DIR%"\build
-set TBBROOT=%PREFIX%
 set CMAKE_PREFIX_PATH=%PREFIX%
+if %dnnl_cpu_runtime%==tbb (
+    set TBBROOT=%PREFIX%
+    set DNNL_CPU_RUNTIME=TBB
+    )
+if %dnnl_cpu_runtime%==omp set DNNL_CPU_RUNTIME=OMP
+if %dnnl_cpu_runtime%==threadpool set DNNL_CPU_RUNTIME=THREADPOOL
```
Comment (xhochy) [inline recipe/bld-onednn.bat:11]: Use `[…]` quotes on Windows for better stability.
```suggestion
if [%dnnl_cpu_runtime%] == [tbb] (
    set TBBROOT=%PREFIX%
    set DNNL_CPU_RUNTIME=TBB
    )
if [%dnnl_cpu_runtime%] == [omp] set DNNL_CPU_RUNTIME=OMP
if [%dnnl_cpu_runtime%] == [threadpool] set DNNL_CPU_RUNTIME=THREADPOOL
```

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (igorsafo) [reply recipe/conda_build_config.yaml]: thank you, fixed.

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (igorsafo) [reply recipe/conda_build_config.yaml]: Fixed

File: recipe/bld-onednn.bat (hunk 3 above)
Comment (igorsafo) [reply recipe/bld-onednn.bat]: Added brackets, thanks!

**Comment by igorsafo** [comment]
@conda-forge-admin, please rerender

**Comment by igorsafo** [comment]
@conda-forge-admin, please restart ci

**Comment by igorsafo** [comment]
@conda-forge-admin, please restart ci

**Comment by cfRod** [comment]
> Is there any use case for oneDNN-threadpool on MacOS? Tensorflow?

Hi @igorsafo , not that we know of.

**Pull request merged**